### PR TITLE
Add sicmutils.env.sci

### PR DESCRIPTION
**Pull request by mk**
First stab at adding a sci environment. All public functions from `sicmutils.env` should work, plus the `literal-function` macro. `with-literal-functions` might work, but I haven't tested it yet. Could not yet figure out how to get `let-coordinates` and `using-coordinates` working. 

So far just tested with Clojure. Think I should at least add some basic tests before merging this.

* [x] Create sicmutils.env.sci with public fns
* [x] Add `literal-function` and `with-literal-functions` macros
* [x] Add tests
* [x] Get ` let-coordinates` and `using-coordinates` working
* [x] Test from ClojureScript

Supersedes #199.

### CODE REVIEW

**Comment by mk** [comment]
@sritchie I made good progress here, also thanks to the generous help of @borkdude. Adding more namespaces should be fairly trivial now.

**Comment by sritchie** [comment]
@mk, I'm catching up post (inter?) holidays, and this is my next focus!

**Comment by mk** [comment]
@sritchie great, let me know if anything is unclear. Is the ci coverage failure a hiccup? Regarding tests I'm unsure if it's worth it to add them now. Like you said, we'd ideally run all tests in sci but that can probably come as a follow-up PR?

**Comment by sritchie** [comment]
Adding a note to myself here to check on a potential issue with "vars" that I noticed in SCI... if we don't define the environment using vars (vs fns), I think no one can access docstrings inside the Nextjournal environment. But if we DO, I noticed that sci does not seem to resolve vars into their attached fns or bindings before calling a fn.

We have a bunch of machinery that looks at the actual object to determine its type for generic dispatch, so this would be a problem.

@borkdude, does this sound familiar?

**Comment by borkdude** [comment]
@sritchie 

> But if we DO, I noticed that sci does not seem to resolve vars into their attached fns or bindings before calling a fn.

I'm not sure what you mean here, but if you can demonstrate this problem using a minimal repro, I'd be happy to look into it.

File: src/sicmutils/env.cljc [hunk 1]
```diff
@@ -83,10 +84,15 @@
      `(af/literal-function ~f ~sicm-signature)))
   ([f domain range] `(af/literal-function ~f ~domain ~range)))
 
-(defmacro with-literal-functions
-  [& args]
+(defmacro with-literal-functions [& args]
   `(af/with-literal-functions ~@args))
 
+(defmacro let-coordinates [& args]
```
Comment (sritchie) [inline src/sicmutils/env.cljc:90]: @littleredcomputer , I realized that these hadn't been exposed to self-hosted cljs, ie, to maria... so, adding them here vs the import below.

**Comment by sritchie** [comment]
@mk , thank you for this!! I'll merge once the tests pass and get the CHANGELOG updated.

**Pull request merged**